src/frequenz/dispatch/_actor_dispatcher.py

```diff
@@ -155,6 +155,12 @@ def new_receiver(self) -> Receiver[Dispatch]:
             """
             return self._channel.new_receiver()
 
+        def cleanup(self) -> None:
+            """Clean up the retry manager."""
+            for task in self._tasks:
+                task.cancel()
+            self._tasks.clear()
```

Comment:
Technically this is not enough, you also should `await` for the tasks, in case they raised an exception for example, otherwise exceptions are swallowed. This is why I think it would be best to convert this class to a proper `BackgroundService`.